static/js/message_edit.js

```diff
@@ -884,22 +881,29 @@ export function save_message_row_edit($row) {
 
     show_message_edit_spinner($row);
 
-    if ($edit_content_input.attr("readonly") !== "readonly") {
+    const $edit_content_input = $row.find(".message_edit_content");
+    const can_edit_content = $edit_content_input.attr("readonly") !== "readonly";
+    if (can_edit_content) {
         new_content = $edit_content_input.val();
         content_changed = old_content !== new_content;
         changed = content_changed;
     }
 
-    if (message.type === "stream") {
-        new_topic = $row.find(".message_edit_topic").val();
+    const $edit_topic_input = $row.find(".message_edit_topic");
+    const can_edit_topic = message.is_stream && $edit_topic_input.attr("readonly") !== "readonly";
+    if (can_edit_topic) {
+        new_topic = $edit_topic_input.val();
         topic_changed = new_topic !== old_topic && new_topic.trim() !== "";
+    }
 
-        if (can_edit_stream) {
-            const $dropdown_list_widget_value_elem = $(`#id_select_move_stream_${message_id}`);
-            new_stream_id = Number.parseInt($dropdown_list_widget_value_elem.data("value"), 10);
-            stream_changed = new_stream_id !== old_stream_id;
-        }
+    const can_edit_stream =
+        message.is_stream && settings_data.user_can_move_messages_between_streams();
```

Comment:
I reverted this change to use `user_can_move_messages_between_streams` again, since we simply hide the stream edit widget when user doesn't have permission. Also, this is much faster than checking `:visible` via jQuery.

Reply:
The speed doesn't matter -- we're not doing this in a loop or anything. But sure, this is fine; I was just hoping to have the code paths be obviously parallel.